source/crud/tests/unified/insertOne-dots_and_dollars.yml

```diff
@@ -0,0 +1,241 @@
+              command:
+                insert: *collection0Name
+                documents:
+                  - *dollarPrefixedKeyInId
+    outcome: *initialData
+
+  - description: "Inserting document with dotted key in _id on 3.6+ server"
+    runOnRequirements:
+      - minServerVersion: "3.6"
+    operations:
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: &dottedKeyInId { _id: { a.b: 1 } }
+        expectResult:
+          insertedCount: 1
+          insertedId: { $$unsetOrMatches: { a.b: 1 } }
+    expectEvents: &expectEventsDottedKeyInId
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                insert: *collection0Name
+                documents:
+                  - *dottedKeyInId
+    outcome:
+      - collectionName: *collection0Name
+        databaseName: *database0Name
+        documents:
+          - *dottedKeyInId
+
+  - description: "Inserting document with dotted key in _id on pre-3.6 server yields server-side error"
+    runOnRequirements:
+      - maxServerVersion: "3.4.99"
+    operations:
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: *dottedKeyInId
+        expectError:
+          isClientError: false
+    expectEvents: *expectEventsDottedKeyInId
+    outcome: *initialData
+
+  - description: "Inserting document with DBRef-like keys"
+    operations:
+      - name: insertOne
+        object: *collection0
+        arguments:
+          # Note: an incomplete DBRef document may cause issues loading the test
+          # file with an Extended JSON parser, since the presence of one DBRef
+          # key may cause the parser to require others and/or enforce expected
+          # types (e.g. $ref and $db must be strings).
+          #
+          # Using "$db" here works for libmongoc so long as it's a string type;
+          # however, neither $ref nor $id would be accepted on their own.
+          #
+          # See https://github.com/mongodb/specifications/blob/master/source/extended-json.rst#parsers
+          document: &dbrefLikeKey { _id: 1, a: { $db: "foo" } }
+        expectResult: *insertResult
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                insert: *collection0Name
+                documents:
+                  - *dbrefLikeKey
+    outcome:
+      - collectionName: *collection0Name
+        databaseName: *database0Name
+        documents:
+          - *dbrefLikeKey
+
+  # Note: the description refers to "pre-5.0" purely for consistency with the
+  # related replaceOne test. Document validation on [3.6, 5.0) server versions
+  # is less strict for insert than it is for replacement documents so this test
+  # actually requires a pre 3.6-server.
+  - description: "Unacknowledged write using dollar-prefixed or dotted keys may be silently rejected on pre-5.0 server"
```

Comment:
The title here also aligns with the original goal from the scope:

> Document that unacknowledged writes using dollar-prefixed or dotted keys may be silently rejected by pre-5.0 servers.

Alternatively, I could swap the document for `*dollarPrefixedKeyInId` since that fails everywhere (pre- and post-5.0). In that case we could probably remove this comment entirely.

Reply:
The explicit failure of a dollar-prefixed key in ID might show this silent rejection better and allow you to remove the comment, but I don't have a strong preference.